app/js/lib/speech/synthesis.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+const p = Object.freeze({
+  // Properties
+  synthesis: Symbol('synthesis'),
+  supportsSynthesis: Symbol('supportsSynthesis'),
+  voice: Symbol('voice'),
+  initialised: Symbol('initialised'),
+
+  // Methods
+  populateVoice: Symbol('populateVoice'),
+});
+
+export default class SpeechSynthesis {
+  constructor() {
+    const synthesis = window.speechSynthesis;
+
+    this[p.initialised] = false;
+    this[p.supportsSynthesis] = !!synthesis;
+
+    if (this[p.supportsSynthesis]) {
+      this[p.synthesis] = synthesis;
+      this[p.populateVoice]();
+      this[p.synthesis].onvoiceschanged = this[p.populateVoice].bind(this);
+    } else {
+      this[p.synthesis] = null;
+    }
+
+    this[p.voice] = null;
+
+    Object.seal(this);
+  }
+
+  /**
+   * Speak a text aloud.
+   *
+   * @param {string} text
+   */
+  speak(text = '') {
+    if (!text) {
+      return;
+    }
+
+    const utterance = new SpeechSynthesisUtterance(text);
+
+    if (this[p.voice]) {
+      utterance.voice = this[p.voice];
+    }
+    utterance.lang = 'en';
```

Comment:
Yes, but there's so much to do to be really multilingual. Some crucial parts of the system are missing for languages other than English (date parser...). Do you think it's still worth doing it?